### DISCUSSION TITLE
error reporting venues

### DISCUSSION THREAD

**Discussion by markfirmware** [alpine-collective/alpinejs-devtools#146] [Remnant]
This post is about number three - if you can veto it I will be just as happy as pursuing it. I had to get this out of my system.

1. the log
1a. a view in an external tool, such as a warnings tab in devtools
2. in-band where there is a conditional div that presents errors (taken from some global list) so that rhey are right in your face without looking at a log
3. mutate the source dom with an error attribute - in the face of the coder
4. anything not above

Number three: If an html element has an error, maybe add an attribute x-see-log-1 where 1 means this is the first detected error. Then repeat the marker x-see-log-1 inside the log (to easily find it) and the log provides the full error report.

A similar approach might be an attribute x-error with its value being the error message.

The error pass would delete stale error attributes.

Mutating the source html might be objectionable, I do not know.

**Comment by HugoDF** [comment]
Have you seen https://github.com/alpine-collective/alpinejs-devtools/pull/140?

We're adding a "warnings" tab to the devtools.

I'm not keen on changing the user's HTML it's the kind of thing that can cause issues down the line (eg. if running an app with devtools enabled changes its behaviour)

**Reply by markfirmware** [reply]
I added 1a (warnings tab) to the list
